src/pages/settings/Security/TwoFactorAuth/Steps/VerifyStep.js

```diff
@@ -32,24 +35,45 @@ const defaultProps = {
 
 function VerifyStep({account, session}) {
     const {translate} = useLocalize();
+    const navigation = useNavigation();
 
     const formRef = React.useRef(null);
 
-    const {setStep} = useTwoFactorAuthContext();
-
     useEffect(() => {
         Session.clearAccountMessages();
+
+        if (account.requiresTwoFactorAuth) {
+            Navigation.navigate(ROUTES.SETTINGS_2FA.ENABLED, CONST.NAVIGATION.TYPE.FORCED_UP);
+            return;
+        }
+
+        if (!account.codesAreCopied) {
+            Navigation.navigate(ROUTES.SETTINGS_2FA.CODES, CONST.NAVIGATION.TYPE.FORCED_UP);
+            return;
+        }
+
+        formRef.current.focus();
```

Comment:
It's breaking the animation transition - please use `useAutoFocusInput` hook

Reply:
Ah, I was wondering why the animation was sometimes okay and sometimes it wasn't. Should be good now!